### ISSUE TITLE
l10n_de

### ISSUE THREAD

**Issue by lap-odoo** [odoo/odoo#18571]
Dear Odoo Partners,

At Odoo, a lot of effort is put into improving the accounting localizations for each country and also to maintain these accounting localizations up-to-date. To achieve that, we will need your help.

One github issue will be created per accounting localization (e.g. l10n_be for Belgium). We expect from you to make suggestions about what should be improved in the specific accounting localization. Do not hesitate to share files with detailed explanations.

As soon as changes will be implemented, you will be informed and will have the opportunity to test it and share your feedback.

This github issue will also be useful to keep you informed about all the ongoing improvements concerning a specific accounting localization.

**Comment by lap-odoo** [comment]
Dear Partners, 

We are already working on: 
- Improved CoAs (SKR03 & SKR04)
- Improved Taxes
- Improved Balance Sheet (SKR03 & SKR04)
- Improved P&L (SKR03 & SKR04)
- Business Assessment Report
- VAT Report
- Intrastat Report
- Datev Export

As soon as this will be ready, I'll inform you so that you can test it. 

Thanks for your help

**Comment by intero-chz** [comment]
@lap-odoo what is meant by "Datev Export". Could you please get a bit more detailed about about what is done already?

**Comment by OSevangelist** [comment]
The new French/German and FERD/FNFE based invoicing reports shall be supported. An OCA module for FERD does however exist already

http://blog.docuware.com/product-news/germany-and-france-single-e-invoicing-format
http://www.ferd-net.de/upload/PDF/Wild_Sautereau_ZUGFeRD_A_Hybrid_Format__for_Electronic_Invoices_in_Europe.pdf

**Comment by OSevangelist** [comment]
In Germany we have a concept which is called the "ISTVERSTEUERUNG" it basically means that as long as the company has not have a turnover of 600.000€ (500.000€ before 2016) or more in the previous year, taxes (as per the VAT reports) need to be payed only if the invoice has been already paid. Otherwise taxes need to be payed as soon as the invoice is issued. However applying the ISTVERSTEUERUNG is not mandatory but may be requested

**Comment by mncoon** [comment]
@intero-chz it is an easy net-based export where are no booking keys or configuration of auto-tax-accounts.

@OSevangelist from my point of view this already exists and works fine. It is called "Tax Cash Basis" and you can activate it from the configuration. There is also a tax-report which uses this option

**Comment by OSevangelist** [comment]
@mncoon did you check whether this function / configuration works in line with the german accounting standard ? I didn't so far but i'll give it a try these days.

**Comment by intero-chz** [comment]
@mncoon so it isn't enough for most of the german datev users. :-1:

**Comment by OSevangelist** [comment]
As of the beginning of 2018 we need far more than even the traditional DATEV export including auto-tax accounts and the like. New GoBD requires SME to allow L1-L3 access to Odoo which requires (special read only rights) as well as full XML/CSV export of ALL business relevant transactions for statistical purposes.

**Comment by mncoon** [comment]
@intero-chz this is enough for those users who do the whole accounting in Odoo, which is possible to do. The export can be sent to the tax-office for reporting and financial statement at the end of the fiscal year. 

@OSevangelist thanks for that hint. a read-only user for every app is necessary. It culd be setted manually, but it is very inconvenient. Maybee you can go a little bit deeper and explain @lap-odoo what is exactly necessary.

**Comment by OSevangelist** [comment]
@intero-chz / @mncoon Non of our customers uses primarily Odoo for accounting and i would really avoid to do so as long as the GoBD compliance regulations are not fully implemented in Odoo (we are far from that)

@lap-odoo / @mncoon As for the GoBD: This reaches far beyond only standard accounting localization requirements as it involves document management, archiving and full traceability and audibility of the systems. GoBD integrates prior German compliances laws (GdPDU and GoBS) and details them further with regard to IT. However, the text base is not available in English but only in German (and its real law but legally binding (in Germany they call it Durchführungsverordnung)). Moreover no leading cases have further detailed the "termini technici" used in the document. I could easily fill a half day workshop on the requirements put in this document (and my interpretation) but i am officially not allowed to do so since we have a protected job description in Germany (Steuerberater) that is the only role that is officially allowed to consult and help people to implement the (law). So what i want to make clear here is, that we @lap-odoo is proposing to do complemented by what we suggest is only the minimum accounting we need for Germany but without full compliance to GoBD i would not consider it a fully localized accounting alternative as of now (and i think this should also not be the target of the first iteration)

http://www.bundesfinanzministerium.de/Content/DE/Downloads/BMF_Schreiben/Weitere_Steuerthemen/Abgabenordnung/Datenzugriff_GDPdU/2014-11-14-GoBD.pdf;jsessionid=87CBA08600DBD5FB5CAEA7EAB46CAAEE?__blob=publicationFile&v=3

**Comment by lap-odoo** [comment]
@intero-chz : Regarding the Datev Export, you can already test it on the master-l10n-de-new-skr-csn enterprise branch on our runbot. There is a button for the Datev Export on the general ledger report. 

@OSevangelist : For the FERD/FNFE e-invoicing, I'll have a look at this. Apparently, that's quite new and from what I know it has not been requested by any of our users until now. That's not our big priority. 

Regarding the ISTVERSTEUERUNG, @mncoon is right, what you define seems to be our cash basis feature. Here is some documentation about it : https://www.odoo.com/documentation/user/11.0/accounting/others/taxes/cash_basis_taxes.html. I hope this will help. 

I found some articles about the GoBD in English. We have already done some work in that direction for l10n_fr. It seems to be more or less similar. We will investigate on that. 

We first would like to finish that first iteration. I'll inform you when it is ready to be tested.

**Comment by OSevangelist** [comment]
Hello Laura, yes FERD/FNFE is still quite new but it will be legally binding for all those who want to make bid on public offers and for the public organisations by 2019 (see http://www.kommunal-edv.de/wissen/it-technik/schnittstellen/zugferd/). However, the matter is covered in an OCA Modul and might better reside there for the time being. According to yours and the hint of @mncoon i didn't check the "cash based" feature but at least it sounds like it is implementing what is required by "ISTVERSTEUERUNG". According to the GoBD we have done a whole implementation individually for one of our older customers (still running on v7) but we might lift that to v8+ and contribute that to the OCA soon, so probably no need to invest here to much either. One thing poppt up just now is the translation that is still somehow weired in many occasions in v10 for the German part at least. It would certainly be helpful to invest a bit there

**Comment by BT-astauder** [comment]
The bank data of all banks can be found here:
https://www.bundesbank.de/Redaktion/EN/Standardartikel/Tasks/Payment_systems/bank_sort_codes_download.html?https=1

There are two things to mention:
- Disclaimer, that official bank sort code update service for payment service providers is available via secure download (not needed in my eyes).
- Data only valid for 3 months. There are already changes since 5 June (4 pages only available as PDF). So kind of an update mechanism (cron job) is needed to update the data.

**Comment by tv-openbig** [comment]
I have created a sheet with a quick compliance Check of german "GoBD" accounting rules :
https://docs.google.com/a/big-consulting.net/spreadsheets/d/1IB00J882mZCpWASsTLyAs-M1u5AXxlQCSQYaHprlGQg/edit?usp=sharing

Most improvements described by @lap-odoo seems to go in the right direction 
(especially proper reports).

As a result of my personal compliance check above i also see a gap to 
fulfill the requirement "Automated evaluability through read-only data 
access by the financial authority". Currently Odoo doesn't support 
read-only data access as @OSevangelist and @mncoon already mentioned. 

This 3-rd-party module module may fill this gap: 
https://www.odoo.com/apps/modules/10.0/accounting_auditor/

In practise a tax auditor will ask you about IDEA data archive at first.
Currently Odoo doesn't support that. It is in fact possible 
to create those files by hand, however advanced accounting skills 
and knowledge about the IDEA format is required.

Less important in context of GoBD, however a good opportunity to add real value 
to german Odoo accountants:   "Elsteronline" xml export file to declare monthly 
or quarterly the VAT taxes. "Elsteronline" is a webinterface provided by german 
tax authorities. To avoid manual data entries the webapp also provides a way to 
import .xml file.

@lap-odoo Regarding datev_export: Indeed it could be interesting to take a look 
at your datev_export module. Just from the pull request here (https://github.com/odoo/odoo/pull/17447/), without access to the corresponding 
runbot instance and without any context of the exact DATEV format 
it is impossible to contribute in any way. Could you ask @qdp-odoo to 
provide more details ? 

Last not least: Which of the new l10n_de features will be in Odoo CE ?

**Comment by qdp-odoo** [comment]
I quite don't get the question, sir

**Comment by OSevangelist** [comment]
@initos we have made a couple of modules for an old Odoo v7 customer that does read only (for L1 and L2 access) as well as an export that exports everything in IDEA XML syntax. We did not update that to V8+ so far but we created it in a way that it is easily portable. I am thinking of putting this into the OCA but I suggest to have a narrow l10n standard Odoo module and than something that is more elaborated in the OCA what do the others think

**Comment by tv-openbig** [comment]
@qdp-odoo 
My question is about the DATEV export you plan to integrate in l10n_de
1. ) Which DATEV format do you plan to support ?
As some different formats exist:
DATEV-format (.ascii)
DATEV XML

2.) How is the coverage of the datev export ?
- Invoices (Customer, Supplier, Refunds) ?
- Cash Accounting (Account moves in cash journal) ?
- Attachments to Invoices ?

3.) How is it possible to test development state on runbot ?

**Comment by mncoon** [comment]
@tv-openbig I had the chance to check and think can answer your questions:
1. the format is ASCII (not DATEV Standard, must be configured)
2. the export covers all moves in an easy export. For more details please have a look on my above comment:

> it is an easy net-based export where are no booking keys or configuration of auto-tax-accounts. For >example: income move is one line and tax move is also one line in the export.
>
>Amount | Debit | Credit
>100,00 | Debitor Account | Income Account
>19.00 | Debitor Account | Tax Account

3. at the moment there is no possibility for a check. I think @lap-odoo will inform us if there is a new test

**Comment by lap-odoo** [comment]
For the read-only data access, I agree, that's a good suggestion. I've added this in our backlog. 

@tv-openbig : I've rebuilt the branch master-l10n-de-new-skr-csn on odoo runbot (odoo-dev/enterprise). It's still in testing, once it will be done you should be able to test the Datev export (there is a button on the general ledger). 

Once everything will be ready, you will all be informed that it can be tested.

**Comment by D4N** [comment]
I have a suggestion concerning the skr04 chart of accounts. If you install it, the accounts which include taxes have them not set by default (e.g. account 4400 should have "19% Umsatzsteuer skr04" set as the default tax). I think not everyone needs this linking, if they don't have to pay certain taxes (please don't quote me on that, I am not an accountant) but having an option to enable the default taxes would be nice.

Also, my colleague told me, that the skr04 chart of accounts is missing the fiscal position for customers & vendors from Germany (would be "Kunde Deutschland" & "Lieferant Deutschland Unternehmen (mit USt-ID) skr04" following the convention of the other fiscal positions). According to her, you have to apply "7/19% Vorsteuer" for a German customer but "7/19% Umsatzsteuer" for a EU customer. If this is correct, it should be added, too. Unfortunately, I cannot confirm this with certainty, as I was unable to find some official documents concerning the fiscal positions.

**Comment by OSevangelist** [comment]
Well by far, most of the German companies in the SME sector (i am aware of) run normal businesses. This means they have to apply 7/19% Umsatzsteuer on sales and to pay 7/9% Vorsteuer on Purchases / Procurements. In that regard the statement  "According to her, you have to apply "7/19% Vorsteuer" for a German customer but "7/19% Umsatzsteuer" is wrong. While trading with European Member Countries it depends on the business model, the volume and the country what taxes apply. For instance if you surpass 100.000€ (don't quote me on that, i am no accountant either) of annual sales from Germany to Austria with end customers (i.e. e-Commerce) you have to apply their 20% Tax instead of the German 19%. So the golden rule. Don't do accounting with Odoo (or any other solution in the market place) without having proper task accountants / consultants at hand ;-)

**Comment by D4N** [comment]
You are right, I don't intend to do the accounting myself (or anyone else in the company). However it would be nice to get as close to the actual rules as possible, especially concerning the linking of the accounts via taxes.

**Comment by csnauwaert** [comment]
Hello guys,

We've just finished reworking the VAT report according to this source: http://www.bundesfinanzministerium.de/Content/DE/Downloads/BMF_Schreiben/Steuerarten/Umsatzsteuer/2016-10-19-Muster-USt-Voranmeldungs-Vorauszahlungsverfahren-2017.pdf?__blob=publicationFile&v=4

We've had to rewrite completely the previous one and we had to add several tags that needs to be set on the taxes in the process. We, however, don't really know which tags should be set on which taxes, so we will need your help to set them.

Here is a file with the current taxes for l10n_de, could you please fill the tags column with the value of the tag(s)? This will help us and prevent each user from configuring their taxes themselves. For example, we know that tax "0% Steuerfreie Leistung EU" goes in grid 21 of the report but we don't really know for all the others.

[tax_tag_incomplete.zip](https://github.com/odoo/odoo/files/1284110/tax_tag_incomplete.zip)

Thanks a lot

Btw: branch master-l10n-de-new-skr-csn is ready for testing (we just miss the association of tags on the taxes for the VAT report and the business assessment report which will be done later)

**Comment by mncoon** [comment]
Hi @csnauwaert, 

I added the missing tags. I used the big numbers form the source which are before the both culums and not the grid line numbers:
http://www.bundesfinanzministerium.de/Content/DE/Downloads/BMF_Schreiben/Steuerarten/Umsatzsteuer/2016-10-19-Muster-USt-Voranmeldungs-Vorauszahlungsverfahren-2017.pdf?__blob=publicationFile&v=4

I also added two new tax rates which are existing since a long time, but missing in Odoo. Basically it is the same as the "Innergem. Erwerb 19%USt/19%VSt", but it is not for goods rather for services.

If you have any questions feel free to ask. If necessary we also can do a little metting over hangouts or something.

Here is the updatet file:
[tax_tag_completed.zip](https://github.com/odoo/odoo/files/1284449/tax_tag_completed.zip)

**Comment by csnauwaert** [comment]
@mncoon Thanks a lot, this is exactly what we needed :+1:

**Comment by qdp-odoo** [comment]
Le localization improvement have been merged in master (for saas-18) under follwoing commits:
- community: https://github.com/odoo/odoo/commit/09189fa809ac2e28d509d6c4780fc191acece44e
- enterprise: https://github.com/odoo/enterprise/commit/76185755cb8358dbceadf5b8cda800833ce45705

Feel free to test and comment. Thanks a lot for the contribution!

**Comment by BT-astauder** [comment]
In my eyes, the charts of account is not correct. There are no fixed assets assigned.
Also in the balance sheet: "Anlagevermögen" (PNCA) should be before "Sachanlagen" (FA).

**Comment by csnauwaert** [comment]
@BT-astauder Yes it is possible that some informations on the charts of account might not be entirely correct, we are working with the informations we find over internet, could you tell us what accounts  should be of type "fixed assets"?

Thanks a lot

**Comment by tv-openbig** [comment]
Hello,

Fixed Assets SKR03:
005%,006%,007%,008%,009%,01%,02%,03%,04%
Fixed Assets SKR04:
02%,03%,04%,05%,06%,07%

Balance Sheet EE (some errors on „passiv site“):
===================================
Modify:
--------
B. Sonderposten mit Rücklagenanteil -> C. Rückstellungen
Total B. Sonderposten mit Rücklagenanteil -> Total C. Rückstellungen
B.1. -> C.1
B.2 -> C.2
B.3 -> C.3

C. Rückstellungen -> D. Verbindlichkeiten
Total C Rückstellungen -> Total D. Verbindlichkeiten
C.1 -> D.1
C.2 -> D.2
C.3 -> D.3
C.4 -> D.4
C.5 -> D.5
C.6 -> D.6
C.7 -> D.7
C.8 -> D.8

finally add (before „C“):
-------------------
B. Sonderposten mit Rücklagenanteil

Profit & Loss Report (EE) (some typos)
============================

Modify:
3. Andere aktivierte eigenleistungen -> 3. Andere aktivierte Eigenleistungen
4. Sonstige betriebliche erträge -> 4. Sonstige betriebliche Erträge
8. Sonstige betriebliche aufwendungen -> 8. Sonstige betriebliche Aufwendungen
Total 8. Sonstige betriebliche aufwendungen ->Total 8. Sonstige betriebliche Aufwendungen
Total Ergebnis der Gewöhnlichen Geschäftstätigkeit -> Total Ergebnis der gewöhnlichen Geschäftstätigkeit
Steuern vom einkommen und Ertrag -> Steuern vom Einkommen und Ertrag
14. Steuern vom einkommen und Ertrag -> 14. Steuern vom Einkommen und Ertrag
15. Sonstige steuern -> 14. Sonstige Steuern
Total Steuern vom einkommen und Ertrag -> Total Steuern vom Einkommen und Ertrag

**Comment by BT-astauder** [comment]
I may be also helpful to import SKR03 and SKR04 by standard in english and having a german translation (as it is for everything else in odoo). You can find the english version here:
https://www.datev.de/web/de/datev-shop/?query=skr&Berufsgruppe=Steuerberater&Produktart=Arbeitsmaterial%20%26%20Hardware&ArbeitsmaterialHardware=Kontenrahmen%20(SKR)

**Comment by csnauwaert** [comment]
Hello @tv-openbig 

We've made the change on the accounts type for fixed assets. Thanks a lot for the precision
We've also fixes the typos in the P&L report
Both fixes are already deployed.

Unfortunately I did not understand what you meant by changing the Balance Sheet report.

We used the following document that were given to us to write the Balance sheet and P&L (see attachment)
[skr3_4-BS-PL.zip](https://github.com/odoo/odoo/files/1344327/skr3_4-BS-PL.zip)

Do you mean that the currently section named "B. Sonderposten mit Rucklagenanteil" should be renamed "C. Ruckstelling"? If yes, it must be noted that there is no sublevel to the current "B. Sonderposten mit Rucklagenanteil". So I don't understand the "B1 -> C1", "B2 -> C2", "B3 -> C3" thing. And how should we rename the section after D?
Actually my questions should be: is the accounts we've set in the different element of the report wrong? If yes could you tell us which one are wrong?

Thanks a lot


@BT-astauder Actually all the localization we have are directly written in the language of the country, for example Belgian CoA is in French with a translation to Dutch

**Comment by BT-mkramer** [comment]
Hi 
The structure looks very good.
I suggest the following changes for SKR04 as marked in the attached document. In column A is described what can be changed. 
As basic for my suggestion I used the official Kontenrahmen from datev about structuring skr04.
https://www.datev.de/web/de/datev-shop/material/kontenrahmen-datev-skr-04/

[Balance-Sheet and Profit and Loss - SKR04_Update290917.xls.tar.gz](https://github.com/odoo/odoo/files/1344819/Balance-Sheet.and.Profit.and.Loss.-.SKR04_Update290917.xls.tar.gz)

**Comment by tv-openbig** [comment]
@csnauwaert
Indeed the .xls templates you have used are OK.
There was just a bit mismatching in the tree you have created
on the passive site of the balance sheet.

Just some lines have to be changed. Furthermore 
one position change is needed after the suggested 
line modifications (see screenshot below)

<img width="672" alt="bildschirmfoto 2017-10-02 um 10 52 30" src="https://user-images.githubusercontent.com/8348598/31070465-c0b1efe8-a760-11e7-864d-4678e725affe.png">

The suggestion of @BT-mkramer for SKR04 is looking good too.

**Comment by haconsult** [comment]
Hi all,
To my knowledge LIFO is the only valid costing method for tax balance sheets since the new legislation (BilMoG). See EStG § 6 (1) 2a:

_"[...] Steuerpflichtige, die den Gewinn nach § 5 ermitteln, können für den Wertansatz gleichartiger Wirtschaftsgüter des Vorratsvermögens unterstellen, dass die zuletzt angeschafften oder hergestellten Wirtschaftsgüter zuerst verbraucht oder veräußert worden sind, [...]"_

Unfortunately LIFO is no longer available in odoo v11. Are there any plans to re-implement this costing method? How do you handle this limitation right now?

PS: FIFO is still valid for commercial balance sheets (not for tax purposes); AVCO is not allowed, see HGB § 256.

**Comment by OSevangelist** [comment]
Hi, although i am not an tax expert, i would read the cited legal text

"können für den Wertansatz gleichartiger Wirtschaftsgüter des
Vorratsvermögens unterstellen, dass die zuletzt angeschafften oder
hergestellten Wirtschaftsgüter zuerst verbraucht oder veräußert
worden sind, [...]"

as follows

"..., you MAY select LIFO"

for the tax report or AVCO as this should be allowed as well, at least
according to this source

https://www.smartsteuer.de/online/lexikon/v/vorratsvermoegen-lexikon-de
s-steuerrechts/

However, i'd be interested in the official reasoning for taking off an
evaluation method in version 11 as well.

Kind regards

Frederik


Am Sonntag, den 08.10.2017, 07:07 -0700 schrieb haconsult:
> Hi all,
> To my knowledge LIFO is the only valid costing method for tax balance
> sheets since the new legislation (BilMoG). See EStG § 6 (1) 2a:
> "[...] Steuerpflichtige, die den Gewinn nach § 5 ermitteln, können
> für den Wertansatz gleichartiger Wirtschaftsgüter des
> Vorratsvermögens unterstellen, dass die zuletzt angeschafften oder
> hergestellten Wirtschaftsgüter zuerst verbraucht oder veräußert
> worden sind, [...]"
> Unfortunately LIFO is no longer available in odoo v11. Are there any
> plans to re-implement this costing method? How do you handle this
> limitation right now?
> PS: FIFO is still valid for commercial balance sheets (not for tax
> purposes); AVCO is not allowed, see HGB § 256.
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
-- 
Dr.-Ing. Frederik Kramer
Geschäftsführer
        
initOS GmbH
An der Eisenbahn 1
21224 Rosengarten
        
Phone:  +49 4105 56156-12
Fax:    +49 4105 56156-10
Mobil:  +49 179 3901819
        
Email: frederik.kramer@initos.com
Web:   www.initos.com
        
Geschäftsführung:
Dr.-Ing. Frederik Kramer & Dipl.-Ing. (FH) Torsten Francke

Sitz der Gesellschaft: Rosengarten – Klecken
Amtsgericht Tostedt, HRB 205226
Steuer-Nr: 15/200/53247
USt-IdNr.: DE815580155

**Comment by haconsult** [comment]
Hi, thanks for the quick response!

To my understanding (and I am quite sure here), real ("Einzelbewertung") is the default costing method. It is defined in section above:

_"EStG § 6 (1) 2:
[...] Umlaufvermögen) sind mit den Anschaffungs- oder Herstellungskosten oder dem an deren Stelle tretenden Wert, vermindert um Abzüge nach § 6b und ähnliche Abzüge, anzusetzen [...]"_

You **MAY** select LIFO as the **ONLY SIMPLOFICATION ALLOWED** under certain circumstances (section 2a). 

Even your source says this. Look at the last row in the table:
"a) LIFO-Methode, allowed in HR [Handelsrecht → commercial balance] and StR [Steuerrecht → fiscal balance]
b) FIFO-Methode, allowed in HR only"

Regards, Helge

**Comment by haconsult** [comment]
PS: 
I understand AVCO is no longer allowed, neither in commercial balance sheets, nor in fiscal, since it is not mentioned any more in the legislation.

**Comment by OSevangelist** [comment]
Yes, you are right about the default. That's how i also read it and it is written in the law. However, it depends very much on the business case whether somebody is eligible for an "alternate" approach. I think most real cases could (and actually will) opt such an "alternate" approach for simplicity, as explained in the link i attached above.

**Comment by OSevangelist** [comment]
> PS:
> I understand AVCO is no longer allowed, neither in commercial balance
> sheets, nor in fiscal, since it is not mentioned any more in the
> legislation.
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

Well it is still written here

http://einkommensteuerrichtlinien.de/EStR-6-8-Bewertung-des-Vorratsverm%C3%B6gens.html

under Gruppenbewertung. Unless this hasn't been replaced by something newer it should still be valid.

KR Frederik

**Comment by haconsult** [comment]
True, AVCO may be an option. Thanks for the hint. 
However, would still be good to have LIFO as an other option since FIFO is not compliant.

Regards. Helge

**Comment by mncoon** [comment]
@lap-odoo I created two pull request to correct the errors / mistakes on the reports I found so far:
https://github.com/odoo/odoo/pull/19959
https://github.com/odoo/enterprise/pull/1487

If there are questions or some need to discouss feel free to do it.

**Comment by lap-odoo** [comment]
@mncoon Thank you, 

I see that @csnauwaert already took care of your PR.

**Comment by lap-odoo** [comment]
Hi all, 

We will continue working on the l10n_de module. @csnauwaert and me should soon come with other improvements to test. 

Thanks again for your cooperation.

**Comment by tv-openbig** [comment]
Hello,

i have created a PR to update the german accounting chart for Odoo v10:
https://github.com/odoo/odoo/pull/20745

@lap-odoo @csnauwaert : Would you please check the issue with the CLA ?
Unfortunately automatic test of the PR was blocked by wrong CLA file name.
I have changed the CLA file name now and obviously my PR now includes 
the updated CLA in the folder "/doc/cla/individual" (tv-openbig.md).

I would propose to adopt the proposed accounting chart improvements also 
to V11.

By accepting the PR we should have a better and cleaner base to create 
further missing accounting reports  also for Odoo V10 (f.e. taxes).

Thanks in advance

**Comment by lap-odoo** [comment]
@tv-openbig 

Hi, 

We don't merge any new things in V10 as it is a stable version. On stable versions, we only fix bugs. 

We've just finished to improve the CoA for the V11, these changes have been validated by the Odoo partners who participated in this issue. We don't plan to do other changes on the CoA (expect if we notice they are needed to implement reports, etc.). 

Did you check the last things that have been done for l10n_de for the V11 ? 

If you want, you can explain here what you would suggest to change but based on what we have in V11, not V10. 

Thanks

**Comment by tv-openbig** [comment]
Hello,

i have checked and compared the recent changes done on l10n_de modules for V11. The PR includes f.e. the backport of new V11 tags because these tags are relevant to fix the gap of missing accounting reports (Odoo CE) or incomplete accounting reports (Odoo EE).  Therefore i thought it could be a good idea to harmonize the tags  of V10 / V11.

I understand that main intention of stable versions policy is to keep the database schema, which is obviously the case as this PR doesn't include a database schema change.  

Furthermore this PR adds the missing taxes to be able to cover each tax case mentioned by @csnauwaert  already: 

> http://www.bundesfinanzministerium.de/Content/DE/Downloads/BMF_Schreiben/Steuerarten/Umsatzsteuer/2016-10-19-Muster-USt-Voranmeldungs-Vorauszahlungsverfahren-2017.pdf?__blob=publicationFile&v=4

In fact the fixing of missing tax keys included in this PR could be forward-ported also to V11. For sure the V11 is also affected by missing taxes and this PR is the perfect blueprint to fix this leak.  
Another fix  / improvement was to add the reference between accounts and taxes as mentioned  in this thread by  @D4N :

> However it would be nice to get as close to the actual rules as possible, especially concerning the linking of the accounts via taxes.

This account <-> tax linking is deep in the nature of the DATEV accounting charts where they call this functionality "automatic account " (in german: Automatikkonto). If you use these accounts in DATEV accounting software the assignation of tax accounts and tax tags are done automatically just by selection of the right relevant account with automatic tax key feature.  This feature makes it significantly easier to edit the accounts by unexperienced accountants. Therefore we can expect significant usability improvement introduced by acceptance of this PR. Beyond that we can call it a fix of relevant accounts with automatic tax functionality in origin DATEV account charts (all accounts marked with the tag "AV" or "AM").

To give an example of a non covered tax case we could take a deeper look at a supply transaction of a constructor. By just selecting the right account number "3120" (in SKR03 - "5920" in SKR04) we would create two account move lines on two specific account numbers "1787" as well as on "1577" for the both tax lines. Also automatically the right tax tags would be assigned automatically (tag "84" and tag "85"). Finally the target of this PR is to cover all theoretically possible tax positions in the official VAT declaration, which is currently not the case.

Until now it was required to configure missing complex tax "templates" by hand.To be clear this is also the case in current V11. Therefore this linking is actually required, if we want to make life easier for Odoo accountants in germany. **Again i argue that this PR don't change the database schema**. 

This PR  includes also some further changes of mappings related to an account. Most of them have been discussed in this thread (f.e. fixing of account - tag relation, account - reconcile relation). I can't see a reason why these account chart fixes should not be part of V10 as well.

To tell the truth we now have to go further to bridge the reporting gap we are suffering from in Odoo CE as they was removed by introduction of Odoo EE. Therefore we want to develop  two contributions to the relevant OCA repository "l10n-germany":

**Contribution 1:** a module to create german VAT declaration report (inspired by "l10n- netherland" project).
**Contribution 2:** a module to create detailed balance and P&L reports (technically based upon the OCA mis-report builder framework).

To give an idea i attach some screenshots for the tax declaration:

Screenshot 1 (predefined Configuration screen of VAT report):

<img width="421" alt="bildschirmfoto 2017-11-13 um 17 13 55" src="https://user-images.githubusercontent.com/8348598/32743354-6e6cc1d2-c8ac-11e7-8cc9-dd585e1c3f7b.png">


Screenshot 2 (populated pre-view of report to print, with possibility to update and save to keep the VAT history):

<img width="283" alt="bildschirmfoto 2017-11-13 um 17 14 23" src="https://user-images.githubusercontent.com/8348598/32743336-61cfff3e-c8ac-11e7-832f-62182f0c9fc6.png">

Screenshot 3 (print-out of the VAT report):

<img width="371" alt="bildschirmfoto 2017-11-13 um 17 15 31" src="https://user-images.githubusercontent.com/8348598/32743402-91e6ee62-c8ac-11e7-80f6-e217f9fa0694.png">

Finally the result will be a much better german localisation, with all missing accounting 
reports currently restricted to Odoo Enterprise users. Nevertheless by applying these PR also Odoo Enterprise users would benefit from better usability (better coverage of relevant tax cases, better automatic account and tax tag assignation).

**Comment by OSevangelist** [comment]
Hi German l10n followers, hi Odoo,

the reasoning of Thorsten makes perfect sense. Given the importance to support the German market as mentioned by Fabien within his keynote during the Odoo Experience, i would also strongly recommend to merge the PR. This "Automatikkonto" mapping may bei unique to DATEV and the German market but its dramatically easing the work for German accountants and customers.


If it ist necessary you might flag it as a bugfix because business-wise its just that.

So +1 for merge from my side

Frederik

Am 13. November 2017 20:30:07 MEZ schrieb Thorsten Vocks <notifications@github.com>:
>Hello,
>
>i have checked and compared the recent changes done on l10n_de modules
>for V11. The PR includes f.e. the backport of new V11 tags because
>these tags are relevant to fix the gap of missing accounting reports
>(Odoo CE) or incomplete accounting reports (Odoo EE).  Therefore i
>thought it could be a good idea to harmonize the tags  of V10 / V11.
>
>I understand that main intention of stable versions policy is to keep
>the database schema, which is obviously the case as this PR doesn't
>include a database schema change.  
>
>Furthermore this PR adds the missing taxes to be able to cover each tax
>case mentioned by @csnauwaert  already: 
>
>>
>http://www.bundesfinanzministerium.de/Content/DE/Downloads/BMF_Schreiben/Steuerarten/Umsatzsteuer/2016-10-19-Muster-USt-Voranmeldungs-Vorauszahlungsverfahren-2017.pdf?__blob=publicationFile&v=4
>
>In fact the fixing of missing tax keys included in this PR could be
>forward-ported also to V11. For sure the V11 is also affected by
>missing taxes and this PR is the perfect blueprint to fix this leak.  
>Another fix  / improvement was to add the reference between accounts
>and taxes as mentioned  in this thread by  @D4N :
>
>> However it would be nice to get as close to the actual rules as
>possible, especially concerning the linking of the accounts via taxes.
>
>This account <-> tax linking is deep in the nature of the DATEV
>accounting charts where they call this functionality "automatic account
>" (in german: Automatikkonto). If you use these accounts in DATEV
>accounting software the assignation of tax accounts and tax tags are
>done automatically just by selection of the right relevant account with
>automatic tax key feature.  This feature makes it significantly easier
>to edit the accounts by unexperienced accountants. Therefore we can
>expect significant usability improvement introduced by acceptance of
>this PR. Beyond that we can call it a fix of relevant accounts with
>automatic tax functionality in origin DATEV account charts (all
>accounts marked with the tag "AV" or "AM").
>
>To give an example of a non covered tax case we could take a deeper
>look at a supply transaction of a constructor. By just selecting the
>right account number "3120" (in SKR03 - "5920" in SKR04) we would
>create two account move lines on two specific account numbers "1787" as
>well as on "1577" for the both tax lines. Also automatically the right
>tax tags would be assigned automatically (tag "84" and tag "85").
>Finally the target of this PR is to cover all theoretically possible
>tax positions in the official VAT declaration, which is currently not
>the case.
>
>Until now it was required to configure missing complex tax "templates"
>by hand.To be clear this is also the case in current V11. Therefore
>this linking is actually required, if we want to make life easier for
>Odoo accountants in germany. **Again i argue that this PR don't change
>the database schema**. 
>
>This PR  includes also some further changes of mappings related to an
>account. Most of them have been discussed in this thread (f.e. fixing
>of account - tag relation, account - reconcile relation). I can't see a
>reason why these account chart fixes should not be part of V10 as well.
>
>To tell the truth we now have to go further to bridge the reporting gap
>we are suffering from in Odoo CE as they was removed by introduction of
>Odoo EE. Therefore we want to develop  two contributions to the
>relevant OCA repository "l10n-germany":
>
>**Contribution 1:** a module to create german VAT declaration report
>(inspired by "l10n- netherland" project).
>**Contribution 2:** a module to create detailed balance and P&L reports
>(technically based upon the OCA mis-report builder framework).
>
>To give an idea i attach some screenshots for the tax declaration:
>
>Screenshot 1 (predefined Configuration screen of VAT report):
>
><img width="421" alt="bildschirmfoto 2017-11-13 um 17 13 55"
>src="https://user-images.githubusercontent.com/8348598/32743354-6e6cc1d2-c8ac-11e7-8cc9-dd585e1c3f7b.png">
>
>
>Screenshot 2 (populated pre-view of report to print, with possibility
>to update and save to keep the VAT history):
>
><img width="283" alt="bildschirmfoto 2017-11-13 um 17 14 23"
>src="https://user-images.githubusercontent.com/8348598/32743336-61cfff3e-c8ac-11e7-832f-62182f0c9fc6.png">
>
>Screenshot 3 (print-out of the VAT report):
>
><img width="371" alt="bildschirmfoto 2017-11-13 um 17 15 31"
>src="https://user-images.githubusercontent.com/8348598/32743402-91e6ee62-c8ac-11e7-80f6-e217f9fa0694.png">
>
>Finally the result will be a much better german localisation, with all
>missing accounting 
>reports currently restricted to Odoo Enterprise users. Nevertheless by
>applying these PR also Odoo Enterprise users would benefit from better
>usability (better coverage of relevant tax cases, better automatic
>account and tax tag assignation).
>
>
>  
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/odoo/odoo/issues/18571#issuecomment-344030222

-- 
Dr.-Ing. Frederik Kramer
Geschäftsführer
        
initOS GmbH
An der Eisenbahn 1
21224 Rosengarten
        
Phone:  +49 4105 56156-12
Fax:    +49 4105 56156-10
Mobil:  +49 179 3901819
        
Email: frederik.kramer@initos.com
Web:   www.initos.com
        
Geschäftsführung:
Dr.-Ing. Frederik Kramer & Dipl.-Ing. (FH) Torsten Francke

Sitz der Gesellschaft: Rosengarten – Klecken
Amtsgericht Tostedt, HRB 205226
Steuer-Nr: 15/200/53247
USt-IdNr.: DE815580155

**Comment by tv-openbig** [comment]
Hello,

i have just tested on runbot the DATEV export functionality, which is part of the Odoo Enterprise module l10n_de_reports. Unfortunately i can't report a bug via github issues for Odoo EE.

To my opinion the exported file is not correct. I stumbled over it directly with the first three invoices i have recorded. I hope my example will show the developers like @csnauwaert or @lap-odoo clearly what is wrong to enable them to fix the export file.

**My test scenario:**
1. I have used runbot "odoo/enterprise" branch "11.0".
2. I have created a new company "Germany SKR04" to install l10n_de_skr04.
l10n_de_reports is installed automatically already.
3. I have recorded three invoices (1 customer invoice, 2 vendor bills):

**I. Customer Invoice:**

<img width="1208" alt="invoice1_automatic_account" src="https://user-images.githubusercontent.com/8348598/32835302-e20eb2e4-ca05-11e7-94bb-e50b13b03f31.png">

**II. Vendor Bill 1**

<img width="1219" alt="bill1_automatic_account" src="https://user-images.githubusercontent.com/8348598/32835332-f5f00556-ca05-11e7-8978-263b55eefa20.png">

**III. Vendor Bill 2**

<img width="1206" alt="bill2_no_automatic_account" src="https://user-images.githubusercontent.com/8348598/32835350-034f65c0-ca06-11e7-9417-7bb147eb7088.png">

What is important to understand in the context of DATEV is the fact that the touched account 
in I. and II. is an automatic account in DATEV. In III. i have used a non automatic account.

4. I opened the general ledger report

<img width="1237" alt="general_ledger_before_datev_export" src="https://user-images.githubusercontent.com/8348598/32835443-6c11dd40-ca06-11e7-88f4-a869c44c42a0.png">

5. I clicked on the button "Export DATEV (csv)" to receive the following file:

<img width="820" alt="datev_export_file" src="https://user-images.githubusercontent.com/8348598/32835540-c11ec884-ca06-11e7-8e16-600c1c34db47.png">

In the spreadsheet Odoo produces the fields A1:K10, which is to my opinion wrong format.

**What would i expect instead ?**

I would expect the fields A15:K18 in the spreadsheet.

<img width="1056" alt="general_ledger_datev_export_corrected" src="https://user-images.githubusercontent.com/8348598/32835825-ec3cd0aa-ca07-11e7-906e-3057a6a88d63.png">


I have added the corrected lines as i understand the described format mentioned in the source code of the module l10n_de_reports (Odoo EE):
http://www.datev.de/dnlexom/client/app/index.html#/document/1036228/D103622800029

The comment column in the spreadsheet could help to match the lines of the current wrong exported lines to the corrected lines.

**Comment by lap-odoo** [comment]
Hi, 

Regarding your PR, our developer will have a look at it and see what can be merged in stable versions and what you be kept for the future versions ( @qdp-odoo @csnauwaert ) . 

For the Datev issue, we are working on it.

**Comment by boldt24** [comment]
@OSevangelist Short Info/Kurze Klarstellung:

Umsatzsteuergesetz (UStG) aktuelle Fassung: § 20 Berechnung der Steuer nach vereinnahmten Entgelten

... ein Unternehmer ... dessen Gesamtumsatz (§ 19 Abs. 3) im vorangegangenen Kalenderjahr nicht mehr als 500 000 Euro betragen hat ...

**Comment by boldt24** [comment]
@OSevangelist Short Info/Kurze Klarstellung:

Umsatzsteuergesetz (UStG) aktuelle Fassung: § 20 Berechnung der Steuer nach vereinnahmten Entgelten

... ein Unternehmer, 1. dessen Gesamtumsatz (§ 19 Abs. 3) im vorangegangenen Kalenderjahr nicht mehr als 500 000 Euro betragen hat, oder ...

**Comment by boldt24** [comment]
@D4N If you are not charging VAT, you use another CoA (instead of 4400) 4185
![4185_skr-04-englisch_pdf__seite_12_von_34_](https://user-images.githubusercontent.com/9287107/34342879-ca4f824a-e9be-11e7-8939-33179ff6f62c.jpg)

**Comment by tv-openbig** [comment]
Hello @lap-odoo, @qdp-odoo,

hi german contributors @OSevangelist , @mncoon, @BT-astauder 
and all others interested in this topic.

My PR https://github.com/odoo/odoo/pull/20745 to improve / fix 
the german account chart is still open. You can find detailed
description by opening the PR's link. 

**Please make a decision**, as i would like to publish the 
related tax report module i described with a screenshot series 
in this thread.  I think this new module could be a 
big improvement to simplify the life of german accountants using 
Odoo V10.  Furthermore it could be a good starting point to 
forward it to Odoo V11.

For sure a dependency to official odoo l10n_de modules 
would allow almost all german users to benefit from it.  

Best regards
Thorsten

**Comment by OSevangelist** [comment]
Dear @lap-odoo and @qdp-odoo, 

i think that has something to do with checks still failing because missing
cla.md for Thorsten. As far as i understood he has signed the paper document
since long so this should not block this valuable code from being merged.

Please make it happen soon

thanks a lot

Cheers

Frederik

Am Donnerstag, den 08.03.2018, 16:46 +0000 schrieb Thorsten Vocks:
> Hello @lap-odoo, @qdp-odoo,
> hi german contributors @OSevangelist , @mncoon, @BT-astauder
> and all others interested in this topic.
> My PR #20745 to improve / fix
> the german account chart is still open. You can find detailed
> description by opening the PR's link.
> Please make a decision, as i would like to publish the
> related tax report module i described with a screenshot series
> in this thread. I think this new module could be a
> big improvement to simplify the life of german accountants using
> Odoo V10. Furthermore it could be a good starting point to
> forward it to Odoo V11.
> For sure a dependency to official odoo l10n_de modules
> would allow almost all german users to benefit from it.
> Best regards
> Thorsten
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
-- 
Dr.-Ing. Frederik Kramer
Geschäftsführer
        
initOS GmbH
An der Eisenbahn 1
21224 Rosengarten
        
Phone:  +49 4105 56156-12
Fax:    +49 4105 56156-10
Mobil:  +49 179 3901819
        
Email: frederik.kramer@initos.com
Web:   www.initos.com
        
Geschäftsführung:
Dr.-Ing. Frederik Kramer & Dipl.-Ing. (FH) Torsten Francke

Sitz der Gesellschaft: Rosengarten – Klecken
Amtsgericht Tostedt, HRB 205226
Steuer-Nr: 15/200/53247
USt-IdNr.: DE815580155

**Comment by tv-openbig** [comment]
Hello @lap-odoo , @qdp-odoo , 

i have commited the CLA document as required here:
https://github.com/odoo/odoo/pull/20745/commits/004ce2276f39d976043f16f9ded076afb824c49e

What else is missing to pass the  automated tests ?

**Comment by pedrobaeza** [comment]
Should this still keep opened?

**Comment by tv-openbig** [comment]
@pedrobaeza I have closed the linked OCA issue.

**Comment by qdp-odoo** [comment]
@avw-odoo do you have any use of this "issue"? it's more a placeholder for people to request feature on localizations since we've done everything  so far (I suppose)

**Comment by jannikbecher** [comment]
is there an [ELSTER](https://www.elster.de/eportal/start) integration?

**Comment by Yenthe666** [comment]
I honestly think this should be closed by now. Feel free to correct me if I'm wrong.

**Comment by jannikbecher** [comment]
> I honestly think this should be closed by now. Feel free to correct me if I'm wrong.
I thought this issue is for discussing improving the accounting module for Germany?

**Comment by Yenthe666** [comment]
It's 3.5 years old & nobody from Odoo gave a comment in almost a year plus the person that created this ticket no longer works at Odoo.

**Comment by jannikbecher** [comment]
Ok I see. Thanks for your explanation. Do you have some insights into whether there is an integration for ELSTER?

**Comment by mncoon** [comment]
Hi!

Actually there is no integration for Elster. You have to print the tax 
report and give it manually to ELSTER online.
The duration for this is around 5 minutes per month.


i. A. Mathias Neef
Finanzwirt, Geschäftsführer

T: +49 7826 9663 828
W: https://syscoon.com



Schuttertal - Freiburg - Koci (India)

This message is exclusively intended for its addressee and may contain 
information that is confidential and protected by professional 
privilege. If you are not the intended recipient you are hereby notified 
that any dissemination, copy or disclosure of this communication is 
strictly prohibited by law. If this message has been received by error, 
please immediately notify us via e-mail and delete it

------ Originalnachricht ------
Von: "Jannik Becher" <notifications@github.com>
An: "odoo/odoo" <odoo@noreply.github.com>
Cc: "Mathias Neef" <mathias.neef@syscoon.com>; "Mention" 
<mention@noreply.github.com>
Gesendet: 26.01.2021 16:45:00
Betreff: Re: [odoo/odoo] l10n_de (#18571)

**Comment by jannikbecher** [comment]
Thanks for your fast responses! I was just curious and there is always room for improvement. Maybe I will be able to write my own integration someday.
